polygeist/tools/cgeist/Lib/clang-mlir.h

```diff
@@ -51,10 +51,23 @@ struct LoopContext {
 
 /// Context in which a function is located.
 enum class FunctionContext : unsigned {
-  Host = 0,      ///< Host code
-  SYCLDevice = 1 ///< SYCL device code
+  Host = 0,  ///< Host function
+  Device = 1 ///< Device function
```

Comment:
I'd rather keep the old name, as we might want to add other contexts, e.g., OpenMP, to change how we emit the code.

Reply:
I'm worried we might want to introduce new members to this type, e.g., to add different attributes to the function being emitted depending on the context or to handle namespaces differently. Also, we're only using `SYCLDevice` in the SYCL context, that's why I chose that name.

Reply:
OK given that that is still unclear we can keep the name you want.